cmd/clusterctl/client/tree/discovery.go

```diff
@@ -69,9 +69,9 @@ func Discovery(ctx context.Context, c client.Client, namespace, name string, opt
 	}
 
 	// Adds control plane
-	controlPLane, err := external.Get(ctx, c, cluster.Spec.ControlPlaneRef, cluster.Namespace)
+	controlPlane, err := external.Get(ctx, c, cluster.Spec.ControlPlaneRef, cluster.Namespace)
```

Comment:
And I'm being a little sneaky and fixing the capitalisation here too 🚀

Reply:
I think that we should avoid to call external.Get in case the controlplaneRef is nil given that this is something technically allowed but the API.

Eventually, it would be nice to think how to report this condition to the user; on top of my mind I think we should  add a virtual node with a message reporting that there is no control plane controller defined.
This could have the nice effect to group stand alone control plane machines, if any, but this should be tested...

Reply:
I agree that it makes sense to add some validation before calling `external.Get` and reporting a meaningful message.